conftest.py

```diff
@@ -0,0 +1,79 @@
+import enum
+import inspect
+
+
+def pytest_generate_tests(metafunc):
+    if 'docexample' in metafunc.fixturenames:

```

Comment:
Can you rename `docexample` to `doctest` please? That's usually the common vocabulary.